lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -66,29 +68,32 @@ def self.run(params)
         if release_notes.nil? || release_notes.empty?
           UI.message("⏩ No release notes passed in. Skipping this step.")
         else
-          release = fad_api_client.update_release_notes(release_name, release_notes)
+          release.release_notes = Google::Apis::FirebaseappdistributionV1::GoogleFirebaseAppdistroV1ReleaseNotes.new(
+            text: release_notes
+          )
+          release = client.patch_project_app_release(release.name, release)
         end
 
         testers = get_value_from_value_or_file(params[:testers], params[:testers_file])
         groups = get_value_from_value_or_file(params[:groups], params[:groups_file])
         emails = string_to_array(testers)
         group_aliases = string_to_array(groups)
-        fad_api_client.distribute(release_name, emails, group_aliases)
+        fad_api_client.distribute(release.name, emails, group_aliases)
         UI.success("🎉 App Distribution upload finished successfully. Setting Actions.lane_context[SharedValues::FIREBASE_APP_DISTRO_RELEASE] to the uploaded release.")
 
-        if upload_status_response.firebase_console_uri
-          UI.message("🔗 View this release in the Firebase console: #{upload_status_response.firebase_console_uri}")
+        if release.firebase_console_uri
+          UI.message("🔗 View this release in the Firebase console: #{release.firebase_console_uri}")
         end
 
-        if upload_status_response.testing_uri
-          UI.message("🔗 Share this release with testers who have access: #{upload_status_response.testing_uri}")
+        if release.testing_uri
+          UI.message("🔗 Share this release with testers who have access: #{release.testing_uri}")
         end
 
-        if upload_status_response.binary_download_uri
-          UI.message("🔗 Download the release binary (link expires in 1 hour): #{upload_status_response.binary_download_uri}")
+        if release.binary_download_uri
+          UI.message("🔗 Download the release binary (link expires in 1 hour): #{release.binary_download_uri}")
         end
 
-        Actions.lane_context[SharedValues::FIREBASE_APP_DISTRO_RELEASE] = release
+        Actions.lane_context[SharedValues::FIREBASE_APP_DISTRO_RELEASE] = deep_symbolize_keys(JSON.parse(release.to_json))
```

Comment:
just curious: what does this do / why is this needed?

Reply:
The release was initially represented as a hash with keys that are camel cased [symbols](https://www.rubyguides.com/2018/02/ruby-symbols/) e.g.

```
{
  name: string,
  displayVersion: string,
}
```

The generated API client returns a release object. If you call `to_h` on the release it creates a hash, but the fields are snake case. To ensure it's camel case, I call the `to_json` method which creates a json string with camel case keys and then use `JSON.parse` to convert it back to a hash. While this correctly creates a hash that has camel case keys, the keys in the hash are _strings instead of symbols_.

```
{
  "name": string,
  "displayVersion": string,
}
```

So to ensure we are returning the same type of hash and not breaking folks (e.g. `release[:name]` won't return the name if the keys are actually strings not symbols), we need to _symbolize_ the keys. Does that make sense?

Reply:
Yes. Thank!